src/System.Runtime/tests/System/Reflection/ModuleTests.cs

```diff
@@ -0,0 +1,320 @@
+            var complicatedAttribute = customAttributes[1];
+            Assert.Equal(typeof(ComplicatedAttribute), complicatedAttribute.AttributeType);
+            Assert.Equal(typeof(ComplicatedAttribute).GetConstructors().First(), complicatedAttribute.Constructor);
+            Assert.Equal(1, complicatedAttribute.ConstructorArguments.Count);
+            Assert.Equal(typeof(int), complicatedAttribute.ConstructorArguments[0].ArgumentType);
+            Assert.Equal(1, (int)complicatedAttribute.ConstructorArguments[0].Value);
+            Assert.Equal(1, complicatedAttribute.NamedArguments.Count);
+            Assert.Equal(false, complicatedAttribute.NamedArguments[0].IsField);
+            Assert.Equal("Stuff", complicatedAttribute.NamedArguments[0].MemberName);
+            Assert.Equal(typeof(ComplicatedAttribute).GetProperty("Stuff"), complicatedAttribute.NamedArguments[0].MemberInfo);
+            Assert.Equal(typeof(int), complicatedAttribute.NamedArguments[0].TypedValue.ArgumentType);
+            Assert.Equal(2, complicatedAttribute.NamedArguments[0].TypedValue.Value);
+        }
+
+        [Fact]
+        public void FullyQualifiedName()
+        {
+            Assert.Equal(Assembly.GetExecutingAssembly().Location, Module.FullyQualifiedName);
+        }
+
+        [Fact]
+        public void Name()
+        {
+            Assert.Equal("system.runtime.tests.dll", Module.Name, ignoreCase: true);
+        }
+
+        [Fact]
+        public void Equality()
+        {
+            Assert.True(Assembly.GetExecutingAssembly().GetModules().First() == Module);
+            Assert.True(Module.Equals(Assembly.GetExecutingAssembly().GetModules().First()));
+        }
+
+        [Fact]
+        public void TestGetHashCode()
+        {
+            Assert.Equal(Assembly.GetExecutingAssembly().GetModules().First().GetHashCode(), Module.GetHashCode());
+        }
+
+        [Theory]
+        [InlineData(typeof(ModuleTests))]
+        [InlineData(typeof(PointerTests))]
+        [InlineData(typeof(EventInfoTests))]
+        public void TestGetType(Type type)
+        {
+            Assert.Equal(type, Module.GetType(type.FullName, true, true));
+        }
+
+        [Fact]
+        public void TestToString()
+        {
+            Assert.Equal("System.Runtime.Tests.dll", Module.ToString());
+        }
+
+        [Fact]
+        public void IsDefined_NullType()
+        {
+            var ex = Assert.Throws<ArgumentNullException>(() =>
+            {
+                Module.IsDefined(null, false);
+            });
+            Assert.Null(ex.InnerException);
+            Assert.NotNull(ex.Message);
+            Assert.Equal("attributeType", ex.ParamName);
+        }
+
+        [Fact]
+        public void GetField_NullName()
+        {
+            var ex = Assert.Throws<ArgumentNullException>(() =>
+            {
+                Module.GetField(null);
+            });
+            Assert.Null(ex.InnerException);
```

Comment:
I think this may be going overboard. Checking that it throws ArgumentNullException should be sufficient.